composer.json

```diff
@@ -5,11 +5,11 @@
     "license": "GPL-2.0-only",
     "require": {
         "php": "^7.1",
-        "symfony/http-kernel": "^3.4",
-        "symfony/yaml": "^3.4",
         "doctrine/dbal": "^2.9",
-        "symfony/dependency-injection": "^3.4",
-        "symfony/config": "^3.4"
+        "symfony/config": "^4.2",
```

Comment:
@webhdx is there something specific planned that would disallow supporting Symfony `3.4`? This component is not coupled with anything other from our ecosystem, so it should support `^3.4 || ^4.2`.

Reply:
i agree with @webhdx We had recently lot's of headache in other places because one package was used  by two eZ Platform versions. We will make 0.1 branch and make 0.2 for SF4.

Reply:
Are you aware that this package might be used without eZ Platform? :)
This doesn't make any sense to me, but it's too small to discuss further, so withdrawing my request.

Reply:
@alongosz I' m fully aware. But this packages suites eZ Platform needs first and foremost as it has dependency on this package.

Reply:
Ok. But I'd suggest in this case to make `1.0`. While BC break is allowed in `0.x` packages, it might also cause a headache for us in the future, as stable 2.x-related branches require `^0.1@dev` (which means that 0.2 will be pulled in as well).

Reply:
We can do it, or change meta to pull `~0.1.0@dev`, this is also possible. Not a big deal for us to change meta.

Reply:
I agree with Andrew to release Symfony 4 compatible version as `1.0` because it's a breaking change.

Reply:
> https://github.com/ezsystems/ezpublish-kernel/compare/v6.13.5...master

@alongosz This is incorrect, composer treats every 0.x as major so the only case where it would pull in 0.2 is if it is `~0.1`, ref how we deal with http cache package.

Reply:
So in other word, no rush to do 1.0.0 if the package is not ready for it.

Reply:
> > branches require ^0.1@dev (which means that 0.2 will be pulled in as well).
> 
> @alongosz This is incorrect, composer treats every 0.x as major so the only case where it would pull in 0.2 is if it is `~0.1`, ref how we deal with http cache package.

Hmm... Ok, then it seems composer had a bug in the past. I'm pretty sure we had a problem with `kriswallsmith/buzz` package when it installed `0.17.1` for `^0.16.0` constraint. But now I cannot reproduce it and it looks like it works like you've described.